flexget/components/managed_lists/lists/ombi_list.py

```diff
@@ -0,0 +1,616 @@
+            access_token = self._get_access_token()
+            return {"Authorization": "Bearer %s" % access_token}
+
+        raise plugin.PluginError('Error: an api_key or username and password must be configured')
+
+    def _get_access_token(self):
+        endpoint = "/api/v1/Token"
+        data = {'username': self.config.get('username'), 'password': self.config.get('password')}
+        headers = self.create_json_headers()
+        try:
+            access_token = self._request('post', endpoint, data=data, headers=headers).get(
+                'access_token'
+            )
+            return access_token
+        except (HTTPError, RequestException, ValueError) as e:
+            raise plugin.PluginError('Ombi username and password login failed') from e
+
+    def _request(self, method, endpoint, **params):
+        if not endpoint.startswith('/'):
+            endpoint = '/' + endpoint
+
+        url = self.base_url + endpoint
+
+        headers: Dict[str, str] = params.pop('headers', {})
+        data = params.pop('data', None)
+
+        # add auth header
+        headers.update(self.auth_header.copy())
+
+        response = requests.request(
+            method, url, params=params, headers=headers, raise_status=False, json=data
+        )
+
+        response.raise_for_status()
+
+        return response.json()
+
+    def get(self, endpoint, **params):
+        result = self._request('get', endpoint, **params)
+        return result
+
+    def post(self, endpoint, **params):
+        result = self._request('post', endpoint, **params)
+        return result
+
+    def put(self, endpoint, **params):
+        result = self._request('put', endpoint, **params)
+        return result
+
+    def delete(self, endpoint, **params):
+        return self._request('delete', endpoint, **params)
+
+    @classmethod
+    def create_json_headers(cls):
+        return {'Content-Type': 'application/json', 'Accept': 'application/json'}
+
+
+class OmbiEntry:
+    """Represents a Generic entry returned from the Ombi API."""
+
+    def __init__(
+        self,
+        request: OmbiRequest,
+        entry_type: str,
+        data: Dict[str, Any],
+    ) -> None:
+        self._request = request
+        self.entry_type = entry_type
+        self.data = data
+
+    def mark_requested(self):
+        """Mark an entry in Ombi as being requested."""
+
+        if self.data['requested']:
+            log.verbose(f"{self.data['title']} already requested in Ombi.")
+            return
+
+        log.verbose(f"Requesting {self.data['title']} in Ombi.")
+
+        api_endpoint = (
+            "/api/v1/Request/movie" if self.entry_type == 'movie' else "api/v2/Requests/tv"
+        )
+
+        data = {"theMovieDbId": self.data["theMovieDbId"]}
+
+        headers = self._request.create_json_headers()
+
+        # Maybe we need to try/catch this and throw a plugin exception?
+        response = self._request.post(api_endpoint, data=data, headers=headers)
```

Comment:
Doing some testing with the list, all requests to the API like this should be wrapped in a try statement and if there is an exception do something about it or the whole plugin / task crashes.

IN the case of failure for a lookup write an error and return nothing but continue seems good. IN the case of fatal errors like the auth changes recently commited a faital plugin error should be raised.

Reply:
@bagobones I have refactored the error handling, give it another shot and let me know.

Reply:
Probably won't be able to actually test it till the weekend but the changes look good.

Reply:
I'm going to resolve this conversation but feel free to leave new comments or re-open it.